lib/rucio/core/lock.py

```diff
@@ -66,6 +69,41 @@ def get_dataset_locks(scope, name, session=None):
                'accessed_at': accessed_at}
 
 
+@stream_session
+def get_dataset_locks_bulk(dids, session=None):
+    """
+    Get the dataset locks of a list of datasets or containers, recursively
+
+    :param dids:           List of dictionaries {"scope":scope(type:InternalScope), "name":name,
+                           "type":did type(DIDType.DATASET or DIDType.CONTAINER)}, "type" is optional
+    :param session:        The db session to use.
+    :return:               Generator of lock_info dicts, may contain duplicates
+    """
+
+    from rucio.core.did import list_child_datasets
```

Comment:
At the top `rucio.core.did` is already imported. I think you can just address this as `rucio.core.did.list_child_datasets`